### PR TITLE
Support Apache Impala database #3002

### PR DESCRIPTION
**Pull request by mwullink**
This is the code for adding support for the Impala driver. There are still some tests that are failing. I guess because i'm not using foreign keys. (no impala support) 
The tests cannot find the correct tables?

I need some help with getting the tests to pass.

-  [*] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by camsaul** [comment]
@mwullink Thanks for submitting this.

We'll definitely need some way to test this driver on CI. I'm not really familiar with Impala. How are you running it? Can it be ran locally without too much trouble?

File: docs/administration-guide/databases/impala.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+## Working with Impala in Metabase
+
+Metabase provides a driver for connecting to Impala databases. Under the hood, Metabase uses Impala's JDBC driver; due to licensing restrictions, we can't
```
Comment (salsakran) [inline docs/administration-guide/databases/impala.md]: What license is the driver released under? I can't find a reference at the DL page below.

**Comment by mwullink** [comment]
To test the driver you need a working Impala/Hadoop environment, the easiest way to create one is to use the Cloudera quickstart virtual machine image.
See: http://www.cloudera.com/downloads/quickstart_vms/5-8.html

When the VM is started you have to start Hadoop with a shortcut on the desktop of the VM. After that you have a working Impala database.

**Comment by liulin0103** [comment]
your Impala driver works very well with my Impala cluster. Thank you for your amazing work!

**Comment by feiniao0308** [comment]
@mwullink are you planning to merge the code into master. I'm really expect to see this support. Thanks for the excellent work.

**Comment by lilloraffa** [comment]
Any update on this? I would really love to have official support for Impala! Tnx a lot!

**Comment by gangh** [comment]
Are there any plan to support impala?it is popular now @camsaul

**Comment by Windsmild** [comment]
we also want impala support,please help us @camsaul
Does this pull has any problem?

**Comment by davide-malagoli** [comment]
Is there a way to help with this pull requests @camsaul @tlrobinson @agilliland @mwullink ?
The only test that seems to be failing on circleci is relative to the $ORACLE_JDBC_DRIVER not defined.

Impala is now an Apache Top-Level Project, we will use this in production very soon

https://en.wikipedia.org/wiki/Apache_Impala
https://blogs.apache.org/foundation/entry/the-apache-software-foundation-announces24

**Comment by boredland** [comment]
Hi! Is there an update to that?

**Comment by tlrobinson** [comment]
Hey all.

Thank you for this PR and apologies for the long delay responding. We've recently done some work to support drivers as plugins in Metabase and as of the next release (v0.32) we will be asking driver developers to first publish new drivers in their own repositories (typically named `metabase-*-driver`). If the driver fully passes the driver tests we will link to it from our documentation, and if a driver gains significant usage we will consider merging it into Metabase itself.

Here is some documentation about publishing a driver as a plugin in a separate repository: https://github.com/metabase/metabase/wiki/Writing-a-Driver:-Packaging-a-Driver-&-Metabase-Plugin-Basics#drivers-shipped-as-3rd-party-plugins

Here are two example drivers: https://github.com/metabase/sudoku-driver https://github.com/metabase/crate-driver

We still have some work to do on the documentation. You can follow progress and provide feedback on that here: https://github.com/metabase/metabase/issues/9348

Thank you.

**Pull request closed without merging**